### PR TITLE
EDU-692 Change username to display name

### PR DESCRIPTION
**Pull request by adrianaluchian**
Closes https://educandu.atlassian.net/browse/EDU-692

Screen Shot 2022-07-25 at 10 38 05

### CODE REVIEW

**Comment by adrianaluchian** [comment]
@ahelmberger , so that I don't forget, please have a close look at the German texts and help me correct them ☺️

File: src/components/account-tab.yml [hunk 1]
```diff
@@ -14,8 +14,8 @@ passwordResetEmailSent:
...
+  en: All your public contributions (documents, public rooms, files uploaded to the public storage) will remain on the website and your name will still be displayed as author or contributor. All other data (personal information, private rooms and files uploaded to the private storage) will be removed. You will no longer be able to log into the website.
+  de: Alle Ihre öffentlichen Beiträge (Dokumente, öffentliche Räume und Dateien, die Sie in den öffentlichen Speicher hochgeladen haben), bleiben auf der Website. Ihr Anzeigenamen wird weiterhin als Autor oder Bearbeiter angezeigt. Alle anderen Daten (persönliche Informationen, private Räume und Dateien, die Sie in den privaten Speicher hochgeladen haben), werden gelöscht. Sie können sich dann nicht mehr auf der Website anmelden.
```
Comment (ahelmberger) [inline src/components/account-tab.yml:18]: Anzeigename

File: src/components/displayName-form-item.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+enterDisplayName:
+  en: Please enter a display name
+  de: Bitte geben Sie einen Anzeigenamen an
+displayNameIsTooShort:
+  en: The display name must be at least {length} characters long
+  de: Der Anzeigenamen muss mindestens {length} Zeichen lang sein
+displayNameIsTooLong:
+  en: The display name must be at most {length} characters long
+  de: Der Anzeigenamen muss höchstens {length} Zeichen lang sein
```
Comment (ahelmberger) [inline src/components/displayName-form-item.yml:9]: Anzeigename

File: src/components/login-form.yml [hunk 3]
```diff
@@ -1,12 +1,9 @@
-enterEmailOrUsername:
-  en: Please enter your email address or user name
-  de: Bitte geben Sie Ihre E-Mail-Adresse oder Ihren Benutzernamen an
+enterEmail:
+  en: Please enter your email address
+  de: Bitte geben Sie Ihre E-Mail-Adresse
```
Comment (ahelmberger) [inline src/components/login-form.yml:3]: ... an

File: src/components/pages/login.yml [hunk 4]
```diff
@@ -1,3 +1,10 @@
 forgotPassword:
   en: Forgot your password?
   de: Kennwort vergessen?
+hintContent:
+  en: |
+    The username is no longer supported for logging in.
+    It continues to be shown as the name under which you author website content.
+  de: |
+    Der Benutzername kann nicht mehr für die Anmeldung benutzt werden.
+    Der bleibt als Autorennamen für alle von Ihnen bearbeiteten Website-Inhalt.
```
Comment (ahelmberger) [inline src/components/pages/login.yml:10]: Autorenname

File: src/resources/common.yml [hunk 5]
```diff
@@ -213,9 +216,9 @@ id:
 user:
   en: User
   de: Benutzer
-username:
-  en: User name
-  de: Benutzername
+displayName:
+  en: Display name
+  de: Anzeigenamen
```
Comment (ahelmberger) [inline src/resources/common.yml:221]: Anzeigename

File: src/domain/schemas/user-schema.spec.js [hunk 6]
```diff
@@ -9,14 +10,18 @@ import {
...
+const invalidDisplayNameCases = [
   {
-    description: 'username is shorter than min length',
-    body: { username: 'joedo' }
+    description: 'displayName is shorter than min length',
+    body: { displayName: [...Array(minDisplayNameLength - 1).keys()].join('') }
+  },
+  {
+    description: 'displayName is longer than max length',
+    body: { displayName: [...Array(maxDisplayNameLength + 1).keys()].join('') }
```
Comment (ahelmberger) [inline src/domain/schemas/user-schema.spec.js:24]: As `maxDisplayNameLength` is > 9, the keys in this array start to be 2 characters long each. Probably better would be something like

~~~
Array.from({ length: maxDisplayNameLength + 1 }, () => 'x')

File: src/components/pages/login.yml (hunk 4 above)
Comment (adrianaluchian) [reply src/components/pages/login.yml]: "Er", of course.. I formed the bad habit of saying "Der" in such context all the time.
Thanks, corrected all of the translations you mentioned 🙏

File: src/components/displayName-form-item.yml (hunk 2 above)
Comment (adrianaluchian) [inline src/components/displayName-form-item.yml:9]: Will change these 2 in Nominativ case as well.

Comment (ahelmberger) [reply src/components/displayName-form-item.yml]: Actually, it should be "Der Anzeigename **darf** höchstens {length} Zeichen lang sein" (as oppoesed to "Der Anzeigenamen **muss** mindestens {length} Zeichen lang sein")

**Comment by ahelmberger** [comment]
Looks good!

File: src/components/displayName-form-item.yml (hunk 2 above)
Comment (adrianaluchian) [reply src/components/displayName-form-item.yml]: Thanks for the correction and side-explanation! Updated

File: src/domain/schemas/user-schema.spec.js (hunk 6 above)
Comment (adrianaluchian) [reply src/domain/schemas/user-schema.spec.js]: That's a good point, I changed it as you mentioned and added the positive cases too (which would have caught the setup problem actually, if they had existed before)

**Comment by adrianaluchian** [comment]
> Looks good! Hard to tell if there are some hidden problems.

Even with all my testing, anything hidden would come up later on anyway. Will ask Theresa to test everything.

File: src/domain/schemas/user-schema.spec.js (hunk 6 above)
Comment (ahelmberger) [reply src/domain/schemas/user-schema.spec.js]: Cool, thanks!

**Review by ahelmberger — APPROVED** [review]

**Pull request merged**